packages/snaps-simulation/src/helpers.ts

```diff
@@ -314,6 +325,34 @@ export function getHelpers({
       });
     },
 
+    onNameLookup: async (
+      nameLookupOptions: NameLookupOptions,
+    ): Promise<SnapResponseWithoutInterface> => {
+      log('Requesting name lookup %o.', nameLookupOptions);
+
+      const { ...requestParams } = create(
```

Comment:
Do we need to spread?

Reply:
Yes, in order to get the properties returned from create into that single variable. The rest of the handlers with validation are doing the same thing. It's spread again in the params, because of how name lookup args work.

Other way around is taking each property and assigning conditionally because of `undefined` possibility (`address` and `domain` are optional). A bit more of a mess (and more code) IMO.

Reply:
```suggestion
      const params = create(
```
Could we simply pass it directly like this?

Reply:
Yes, that is also the option in this case. Updated.